main/vlan/vlan.pre-up

```diff
@@ -36,10 +41,8 @@ if ! [ -d /proc/net/vlan ]; then
 	modprobe 8021q
 fi
 
-ip link set dev "$RAW_DEVICE" up
-ip link add link "$RAW_DEVICE" name "$IFACE" type vlan id "$VLAN_ID"
-
-if ! ip link show "$IFACE" >/dev/null; then
-	echo "Failed to create vlan device $IFACE on device $RAW_DEVICE with tag $VLAN_ID"
-	exit 1
+trap "device_creation_error" ERR
+if [ ! -e "/proc/net/vlan/${IFACE}" ]; then
+	ip link set dev "$RAW_DEVICE" up
```

Comment:
I think the `ip link set ... up` should be outside the `/proc/net/vlan/$IFACE` check, in case the interface exists but is `down`.

Reply:
So:
````sh
trap "device_creation_error" ERR
ip link set dev "$RAW_DEVICE" up
ip [ ! -e "/proc/net/vlan/${IFACE}" ]; then
￼	ip link add link "$RAW_DEVICE" name "$IFACE" type vlan id "$VLAN_ID"
fi
````

Reply:
Yes, you're right, thanks!